asmcomp/asmlink.ml

```diff
@@ -203,6 +203,14 @@ let scan_file obj_name tolink = match read_file obj_name with
 
 (* Second pass: generate the startup file and link it with everything else *)
 
+let force_linking_of_startup ppf =
+  let open Cmm in
+  let name = "force_link_of_caml_startup" in
+  Asmgen.compile_phrase ppf
+    (Cdata ([Cglobal_symbol name;
+             Cdefine_symbol name;
+             Csymbol_address "caml_startup"]))
+
```

Comment:
I suspect `Csymbol_address "caml_startup"` is enough, there is no need to give this piece of data a label `force_link_of_caml_startup`.  But you are correct that a mere declaration `Cglobal_symbol "caml_startup"` is not enough, putting the symbol value into the data section is needed to convince the linker.